### ISSUE TITLE
`toString` function / instance for datalog programs

### ISSUE THREAD

**Issue by JonathanStarup** [flix/flix#6872]
The following program shows how to convert any datalog program to a string
```scala
def main(): Unit \ IO = {
    let p = #{A(12). B(42).};
    println(magic(p));
    let _ = List#{1, 2, 3};
    ()
}

def magic(datalog: #{|any}): String = {
    unchecked_cast(datalog as Fixpoint.Ast.Datalog[Boxed]) |> ToString.toString
}

// instance ToString[#{|any}] {
//     pub def toString(x: #{|any}): String = magic(x)
// }

```

Should we add that to std lib somewhere? maybe even in a `ToString` instance? Could that work @mlutze? The instance definition results in an overlapping instance error in `MonoDefs`.

**Comment by mlutze** [comment]
The shape of schemas should allow this. We just need syntax to express a bare schema constructor.

**Comment by magnus-madsen** [comment]
Why should a Datalog program be ToStringable? A lambda function is not ToStringable.

**Comment by magnus-madsen** [comment]
(If we had Debug I could see an argument, but certainly not ToString)

**Comment by JonathanStarup** [comment]
> Why should a Datalog program be ToStringable? A lambda function is not ToStringable.

That was just part of the issue, the primary issue is about any way of printing a Datalog program, a simple function is fine.

And what does lambdas have to do with datalog programs?

**Comment by magnus-madsen** [comment]
I am saying that Datalog programs should not implement ToString.

**Comment by JonathanStarup** [comment]
Yeah that seems sensible. A non-instance function is also fine. This is mostly about the fact that we can easily make a string of a datalog program, which helps debugging a lot. It would be nice to surface that function (magic, from the example)

**Comment by magnus-madsen** [comment]
All of this is gone; like sand in the wind.

The new strategy -- for debugging -- is to use the Debug effect + String or RichString (dealers choice).
